dataframe_show_reader/dataframe_show_reader.py

```diff
@@ -76,19 +82,29 @@ def show_output_to_df(show_output: str, spark_session: SparkSession):
             continue
         line_parts = line.split('|')[1:-1]
         values = [part.strip() for part in line_parts]
-        if types is not None:
-            _cast_types(values, types)
         if column_names is None:
             column_names = values
             continue
-        elif types is None and line.startswith(DATA_TYPE_START_INDICATOR):
-            line = line.replace(DATA_TYPE_START_INDICATOR, '|', 1).rstrip(f'{DATA_TYPE_END_INDICATOR}|') + '|'
-            types = [part.strip() for part in line.split('|')[1:-1]]
-            schema = _get_schema(column_names, types)
-            continue
+        if line.startswith(f'{DATA_TYPE_START_INDICATOR}'):
+            if types is None:
+                line = line.replace(f'{DATA_TYPE_START_INDICATOR}', '|', 1).rstrip(f'{DATA_TYPE_END_INDICATOR}|') + '|'
+                types = [part.strip() for part in line.split('|')[1:-1]]
+                types = [data_type if len(data_type) > 0 else default_data_type
+                         for data_type in types]
+                continue
+            else:
+                raise ValueError('Cannot have more than one data type declaration line.')
+
+        if types is None:
+            types = [default_data_type] * len(column_names)
+
+        _cast_types(values, types)
         row_dict = dict(zip(column_names, values))
         rows.append(Row(**row_dict))
     # Return a DataFrame with the columns in the original order:
+    if types is None:
+        types = [default_data_type] * len(column_names)
```

Comment:
Good idea. I wouldn't have though of this condition if a test case hadn't caught it. I have added this comment:
```
        # This can happen if data types are not specified and no data rows are
        # provided.
```